packages/cloud-cognitive/src/components/DataSpreadsheet/DataSpreadsheetBody.js

```diff
@@ -338,63 +341,88 @@ export const DataSpreadsheetBody = forwardRef(
       ]
     );
 
+    // Builds the empty rows and calls `onDataUpdate` to set the new empty rows
+    // using defaultEmptyRowCount to determine how many empty rows are created.
+    useEffect(() => {
+      if (!rows?.length) {
+        const buildEmptyRows = () => {
+          const emptyRowData = [];
+          [...Array(defaultEmptyRowCount)].map(() => {
+            const emptyCell = {};
+            headerGroups[0].headers.map((header) => {
+              emptyCell[header.id] = null;
+            });
+            emptyRowData.push(emptyCell);
+          });
+          onDataUpdate(emptyRowData);
+        };
+        buildEmptyRows();
+      }
+    }, [rows, headerGroups, defaultEmptyRowCount, onDataUpdate]);
+
+    const RenderEmptyRows = () => {
+      return <div />;
+    };
```

Comment:
This was only necessary because the `<FixedSizeList>` component requires that something always be rendered. However, as soon as we know that `rows` doesn't have any data, we create the empty rows via the useEffect above on line [346](https://github.com/carbon-design-system/ibm-cloud-cognitive/pull/1848/files#diff-44f993b0394c37dc8c3ca1aa905880d04545f5991688d57afb30d9524f7da45dR346) which are the visible empty rows.